File: 1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashDict = dict()
        for num in nums:
            hashDict[num] = nums.index(num)
            
        for i in range(len(nums)):
            rem = target-nums[i]

            if rem in hashDict and hashDict[rem] != i:
                return [i, nums.index(rem)]
        
        return []

# beats 39.13% runtime and 16.24% memory
# https://leetcode.com/problems/two-sum/submissions/1221332929/

File: 1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i=0; i<nums.size()-1; i++)
        {
            for(int j=i+1; j<nums.size(); j++)
                if(nums[i]+nums[j] == target)
                    return {i,j};
        }
        return {};
    }
};

// beats 30.79% runtime and 73.53% memory
// https://leetcode.com/problems/two-sum/submissions/1221320417/
